### PR TITLE
Member expiration frontend

### LINKED ISSUE #284: Member access expiration - UI
# Summary

Currently the group/project members UI doesn't have an expiration field which can be set for a member. Add in functionality so that the group and projects members table has an `Expiration` column which can be used to set/update the expiration date for a membership, as well as, add functionality to the create member modal to set an expiration date.

Related to #230 

1) Add `Expiration` column to the members table
2) Add a `datepicker` input to the members table for each member
3) Add a `datepicker` input to the create member modal
4) Add testing for UI

**Note: The members -> groups tab already has similar functionality so the code for that page can be looked through for examples**

# Acceptance Criteria

Members which have an expires_at set should not be able to access a group or project if their membership has expired

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Fixes #284.
Added date picker input to the members table for each member & create member modal.

## Screenshots or screen recordings

Groups
Screenshot from 2023-12-20 14-22-45
Screenshot from 2023-12-20 14-24-44

Projects
Screenshot from 2023-12-20 14-28-02
Screenshot from 2023-12-20 14-30-29

## How to set up and validate locally
Modify a project/group member:
1. Navigate to a manageable project or group.
2. Click `members` from the navigation menu.
3. Modify the expiration date of a projects/group member.
4. Verify the member was successfully updated.

Create a new project/group member:
1. Click the `Add New Member` button.
2. Fill in the form with & without the expiration date.
3. Verify the member was successfully created.

*Note: Try back dating the expiration date and signing in as the expired user. Verify the expired user does not have access to the project/group.*

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [ ] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/views/groups/members/_member.html.erb [hunk 1]
```diff
@@ -53,22 +55,59 @@
...
-          <%= render UpdatedComponent.new(
-            updated_at: member.updated_at
-          ) %>
+          <%= render UpdatedComponent.new(updated_at: member.updated_at) %>
         <% end %>
       <% end %>
     </div>
   </td>
+  <td class="px-6 py-4">
+    <% if (current_user != member.user && allowed_to?(:update_member?, @namespace)) && !member_source.key?(:inherited_namespace_path) %>
```
Comment (deepsidhu85) [inline app/views/groups/members/_member.html.erb:74]: Logged in as a user with `Maintainer` access level I can update the expiration for an `Owner`. We need to add an `&&` to the first part of the if statement. Take a look at the `access_level` column in this file which has the missing `&&`

image

File: app/views/projects/members/_member.html.erb [hunk 2]
```diff
@@ -70,6 +70,45 @@
       <% end %>
     </div>
   </td>
+  <td class="px-6 py-4">
+    <% if (current_user != member.user && allowed_to?(:update_member?, @namespace)) && !member_source.key?(:inherited_namespace_path) %>
```
Comment (deepsidhu85) [inline app/views/projects/members/_member.html.erb:74]: Same thing applies here as the comment above

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good. Just a few quick comments

File: app/views/groups/members/_member.html.erb (hunk 1 above)
Comment (ksierks) [reply app/views/groups/members/_member.html.erb]: a10c35a94bb03411803d11118e64ef39bdfb6913

File: app/views/projects/members/_member.html.erb (hunk 2 above)
Comment (ksierks) [reply app/views/projects/members/_member.html.erb]: a10c35a94bb03411803d11118e64ef39bdfb6913

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me. Tested and working as expected. Thanks!

**Pull request merged**